conda/plugins/types.py

```diff
@@ -24,11 +25,23 @@ class CondaSubcommand(NamedTuple):
 
 class CondaVirtualPackage(NamedTuple):
     """
-    Conda subcommand entry.
+    A conda virtual package.
 
     :param name: Virtual package name (e.g., ``my_custom_os``).
     :param version: Virtual package version (e.g., ``1.2.3``).
     """
 
     name: str
     version: str | None
+
+
+class CondaSolver(NamedTuple):
+    """
+    A conda solver.
+
+    :param name: Solver name (e.g., ``custom-solver``).
+    :param backend: Callable that will be instantiated as the solver backend.
+    """
+
+    name: str
+    backend: Solver
```

Comment:
```suggestion
    backend: type[Solver
```

Discovered this today while looking at the code in my IDE. Otherwise, it will complain because it expected to receive an instantiated `Solver` object.

Reply:
Yeah, I asked that in https://github.com/conda/conda/pull/11993#discussion_r1019000802

Reply:
@travishathaway This should have been this though? 
```suggestion
    backend: type[Solver]
```

Reply:
Yes, sorry for the syntax error!